nosql/post.json

```diff
@@ -0,0 +1,19 @@
+{
+  "post_id": "Number",
+  "author_id": "Number",
+  "type": "String",
+  "privacy": "String",
+  "text": "String",
+  "hashtags": ["String"],
+  "mentions": ["Number"],
+  "likes": ["Number"],
+  "media": {
+    "images": ["Url"],
+    "videos": ["Url"],
+    "urls": ["Url"]
+  },
+  "comments": ["Number"],
```

Comment:
Does `Number` here represent the ids of the comments? If yes, in my opinion, the main motivation behind including posts/comments/replies in NoSQL is to fetch the post with all its data (comments and replies) in one database query. So I think it would be more convenient to embed the comments/replies as an inner document.
For example
```
{
...
    "comments": [
        {
            "comment_id": "Number",
            ...
            "replies": [
                {
                        ...
                }
            ]
        }
    ]
}
```
What do you think?

Reply:
But then, to edit a reply, we have to edit the whole post?

Reply:
Yes you are right, but editing a reply is not as frequent as viewing a post right? If we were to use this relational pattern, then why not we include it in SQL?

Reply:
@mostafa-abdullah : All numbers are Ids, the thing is non relational database doesn't mean we abandon l relations bl ids(since it is pretty simple). However my point is fearing the scale of the post size. msh aktar.

Reply:
I think NoSQL databases are designed to scale with large documents (Check this blog [here](https://www.mongodb.com/blog/post/storing-large-objects-and-files-in-mongodb)). Imagine that the wall includes X posts, each post has Y comments, then with the multi-collection pattern, we will do X * Y database queries to fetch the posts with its comments and replies, where we would do only a single database query if they are embedded into each other.

Reply:
Tyb I'll put them together :+1:

Reply:
Please refer to our discussion on #nosql on slack if you haven't already :smile: